crates/context/config/src/client/config.rs

```diff
@@ -111,3 +127,46 @@ mod serde_creds {
         }
     }
 }
+
+mod sn_serde_creds {
+    use std::str::FromStr;
+
+    use serde::{Deserialize, Serialize};
+    use starknet_crypto::Felt;
+
+    #[derive(Debug, Deserialize, Serialize)]
+    pub struct Credentials {
+        secret_key: String,
+        public_key: String,
+        account_id: String,
+    }
+
+    impl TryFrom<Credentials> for super::SnCredentials {
+        type Error = &'static str;
+
+        fn try_from(creds: Credentials) -> Result<Self, Self::Error> {
+            'pass: {
+                let public_key_felt = Felt::from_str(&creds.public_key)
+                    .map_err(|_| "Failed to convert public_key to Felt")?;
```

Comment:
_... this patch of code suggests that

Reply:
Updated this also removing .map_err leaving just ? to propagate the felt::FromStrError as it was throwing can't convert to type error